### PR TITLE
Make disk block cache and sync block cache of configurable sizes

### PR DESCRIPTION
**Pull request by jakob223**
Add a few CLI options for `kbfsfuse` to control the size of these caches. Also add a bit of documentation and fix a bit of error handling.

### CODE REVIEW

File: libkbfs/config_local.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ const (
 	// By default, this will be the block type given to all blocks
 	// that aren't explicitly some other type.
 	defaultBlockTypeDefault = keybase1.BlockType_DATA
+
+	// By default, allow 10% of the free bytes on disk to be used in the disk block cache.
+	defaultDiskBlockCacheFraction = 0.10
+
+	// By default, allow 15% of the free bytes on disk to be used in the sync block cache.
+	defaultSyncBlockCacheFraction = 0.15
```
Comment (strib) [inline libkbfs/config_local.go]: Looks like I was wrong about the sync cache size being 15% before.  There seems to be some reporting bug related to my use of the sync cache, that makes the limit different from the working set cache limit.  For now, I say we keep this at `0.10` (change the param default too), and we can change it later if needed.

File: libkbfs/config_local.go [hunk 2]
```diff
@@ -573,6 +583,21 @@ func (c *ConfigLocal) DiskBlockCache() DiskBlockCache {
...
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.diskBlockCacheFraction = fraction
+}
+
+// SetSyncBlockCacheFraction implements the Config interface for ConfigLocal.
+func (c *ConfigLocal) SetSyncBlockCacheFraction(fraction float64) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.diskBlockCacheFraction = fraction
```
Comment (strib) [inline libkbfs/config_local.go]: This should be `syncBlockCacheFraction`, right?

File: libkbfs/config_local.go [hunk 3]
```diff
@@ -594,10 +594,9 @@ func (c *ConfigLocal) SetDiskBlockCacheFraction(fraction float64) {
 func (c *ConfigLocal) SetSyncBlockCacheFraction(fraction float64) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	c.diskBlockCacheFraction = fraction
+	c.BlockCacheFraction = fraction
```
Comment (strib) [inline libkbfs/config_local.go]: `syncBlockCacheFraction`.

Comment (jakob223) [reply libkbfs/config_local.go]: Sigh I'm just being sloppy now..

**Comment by jakob223** [comment]
Would amending my previous commit and force pushing confuse the CI?

**Comment by strib** [comment]
> Would amending my previous commit and force pushing confuse the CI?

No, that's fine.

**Comment by jakob223** [comment]
Also, `go fmt` doesn't touch `init.go`

**Comment by strib** [comment]
Hmm, I get this:

```sh
$ gofmt -w init.go 
$ git diff
diff --git a/libkbfs/init.go b/libkbfs/init.go
index 091f069..97e37b8 100644
--- a/libkbfs/init.go
+++ b/libkbfs/init.go
@@ -202,7 +202,7 @@ func DefaultInitParams(ctx Context) InitParams {
                DiskCacheMode:                  DiskCacheModeLocal,
                DiskBlockCacheFraction:         0.10,
                SyncBlockCacheFraction:         0.10,
-               Mode:                           InitDefaultString,
+               Mode: InitDefaultString,
        }
 }
 
```

**Comment by jakob223** [comment]
Do we have different versions of go or something?
```
[~/go/src/github.com/keybase/kbfs]$ gofmt -w libkbfs/init.go
[~/go/src/github.com/keybase/kbfs]$ git status 
On branch jakob/KBFS-2525-diskblockcache-limit
Your branch is up to date with 'origin/jakob/KBFS-2525-diskblockcache-limit'.

nothing to commit, working tree clean```

**Comment by strib** [comment]
```
$ go version
go version go1.10.4 linux/amd64
```

Same as what CI is running, I think.

**Pull request merged**